include/pybind11/detail/internals.h

```diff
@@ -311,12 +311,23 @@ PYBIND11_NOINLINE inline internals &get_internals() {
     return **internals_pp;
 }
 
+
+struct module_internals {
+  type_map<type_info *> registered_local_types_cpp;
+  std::forward_list<void (*) (std::exception_ptr)> registered_local_exception_translators;
+};
+
+inline module_internals &get_module_internals() {
+  static module_internals locals;
+  return locals;
```

Comment:
This should probably not be called `locals` anymore, since it's not just the local variables anymore?

Reply:
it was all stuff that was local to the module, so seemed appropriate. I'm happy to rename it though. Since it never escapes the context of the function, not sure how much it really matters

Reply:
Doesn't really matter that much, indeed. Just something that caught my eye, since I thought the old name was linked to Python's `locals()`; maybe it isn't, though.

Reply:
I added a `register_local_exception` function, as well.  Someone who is better at c++ template wrangling than me could probably remove the amount of copy-pasta between register_exception and register_local_exception.